exams/pearson/download.py

```diff
@@ -0,0 +1,248 @@
+from exams.pearson.readers import (
+    EACReader,
+    VCDCReader,
+)
+from exams.pearson import utils
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
+)
+
+log = logging.getLogger(__name__)
+
+
+@contextmanager
+def locally_extracted(zip_file, member):
+    """
+    Context manager for temporarily extracting a zip file member to the local filesystem
+
+    Args:
+        zip_file (zipfile.ZipFile): the zip file to extract from
+        member (str|zipfile.ZipInfo): the name or ZipInfo of the member to extract
+
+    Yields:
+        file: the extracted file object
+    """
+    extracted_file_path = zip_file.extract(member, path=settings.EXAMS_SFTP_TEMP_DIR)
+    try:
+        with open(extracted_file_path, 'r') as extracted_file:
+            yield extracted_file
+    finally:
+        if os.path.exists(extracted_file_path):
+            os.remove(extracted_file_path)
+
+
+class ArchivedResponseProcesser(object):
+    """
+    Handles fetching and processing of files stored in a ZIP archive on Pearson SFTP
+    """
+    def __init__(self, sftp):
+        self.sftp = sftp
+
+    def fetch_file(self, remote_path):
+        """
+        Fetches a remote file and returns the local path
+
+        Args:
+            remote_path (str): the remote path of the file to fetch
+
+        Returns:
+            local_path (str): the local path of the file
+        """
+        local_path = os.path.join(settings.EXAMS_SFTP_TEMP_DIR, remote_path)
+
+        self.sftp.get(remote_path, localpath=local_path)
+
+        return local_path
+
+    def filtered_files(self):
+        """
+        Walks a directory and yeilds files that match the pattern
+
+        Args:
+            pattern (re): the regex pattern to mathc the filename on
+        """
+        for remote_path in self.sftp.listdir():
+            if self.sftp.isfile(remote_path) and utils.is_zip_file(remote_path):
+                yield remote_path, self.fetch_file(remote_path)
+
+    def process(self):
+        """Process response files"""
+        try:
+            with self.sftp.cd(settings.EXAMS_SFTP_RESULTS_DIR):
+                for remote_path, local_path in self.filtered_files():
+                    try:
+                        if self.process_zip(local_path):
+                            self.sftp.remove(remote_path)
+                        log.debug("Processed remote file: %s", remote_path)
+                    except SSHException as exc:
+                        raise RetryableSFTPException(
+                            "Error trying to remove file from sftp: {}".format(remote_path)
+                        ) from exc
+                    except:  # pylint: disable=bare-except
+                        log.exception("Error processing file: %s", remote_path)
+                    finally:
+                        if os.path.exists(local_path):
+                            os.remove(local_path)
+        except SSHException as exc:
```

Comment:
What exception would get handles here which wouldn't get handled in the inner except block?